code/ai/aicode.cpp

```diff
@@ -6159,6 +6168,57 @@ int check_ok_to_fire(int objnum, int target_objnum, weapon_info *wip)
 	return 1;
 }
 
+//	--------------------------------------------------------------------------
+//  Returns true if *aip has a line of sight to its current target.
+//	This is a computationally expensive operation, so use sparingly
+int check_los(int objnum, int target_objnum) {
+	vec3d& end = Objects[target_objnum].pos;
+	vec3d& start = Objects[objnum].pos;
+
+	for (int i = 0; i < Highest_object_index; i++) {
```

Comment:
Rather than iterate through all object numbers, it would be better to iterate on `obj_used_list`.